pkg/reconciler/pipelinerun/resources/resultrefresolution_test.go

```diff
@@ -577,7 +601,7 @@ func TestCheckMissingResultReferences(t *testing.T) {
 			pipelineRunState[10],
 		},
 	}, {
-		name:             "Valid: Test successful result references resolution - matrix - array indexing",
+		name:             "Valid: Test successful result references resolution - matrix - whole array replacements",
```

Comment:
Why changing the existing test case not adding a new one? Do we still have array indexing case?

Reply:
We do have an array indexing one still, renamed it to be clearer. https://github.com/tektoncd/pipeline/blob/5ea6147d701d7ae155baf320ac13ee03f2cf3e29/pkg/reconciler/pipelinerun/resources/resultrefresolution_test.go#L292